clients/client-py/README.md

```diff
@@ -308,6 +308,155 @@ taskcluster.fromNow("1 year", dateObj=dateObject1);
 # datetime.datetime(2018, 1, 21, 17, 59, 0, 328934)
 ```
 
+## Library helpers to easily integrate Taskcluster in your project
+
+The Python Taskcluster client has a module `taskcluster.helper` with utilities which allows you to easily share authentication options across multiple services in your project.
+
+Generally a project using this library will face different use cases and authentication options:
+
+* No authentication for a new contributor without Taskcluster access,
+* Specific client credentials through environment variables on a developer's computer,
+* Taskcluster Proxy when running inside a task.
+
+### Shared authentication
+
+The class `taskcluster.helper.TaskclusterConfig` is made to be instancied once in your project, usually in a top level module. That singleton is then accessed by different parts of your projects, whenever a Taskcluster service is needed.
+
+Here is a sample usage:
+
+1. in `project/__init__.py`, no call to Taskcluster is made at that point:
+
+```python
+from taskcluster.helper import Taskcluster config
+
+tc = TaskclusterConfig('https://community-tc.services.mozilla.com')
+```
+
+2. in `project/boot.py`, we authenticate on Taskcuster with provided credentials, or environment variables, or taskcluster proxy (in that order):
+
+```python
+from project import tc
+
+tc.auth(client_id='XXX', access_token='YYY')
+```
+
+3. at that point, you can load any service using the authenticated wrapper from anywhere in your code:
+
+```python
+from project import tc
+
+# Synchronous service class
+queue = tc.get_service('queue')
+
+# Asynchronous service class
+hooks = tc.get_service('hooks', use_async=True)
+
+# You can then use these classes as documented below:
+queue.ping()
+await hooks.ping()
+```
+
+Supported environment variables are:
+- `TASKCLUSTER_ROOT_URL` to specify your Taskcluster instance base url. You can either use that variable or instanciate `TaskclusterConfig` with the base url.
+- `TASKCLUSTER_CLIENT_ID` & `TASKCLUSTER_ACCESS_TOKEN` to specify your client credentials instead of providing them to `TaskclusterConfig.auth`
```

Comment:
Let's mention TASKCLUSTER_PROXY_URL, too, and link to https://docs.taskcluster.net/docs/manual/design/env-vars

Reply:
Done :+1: